### PR TITLE
COOPR-721 Only use provider-hostnames when configured to do so

### PR DESCRIPTION
**Pull request by wolf31o2**

### CODE REVIEW

File: lib/provisioner/worker/plugins/providers/fog_provider/fog_provider.json [hunk 1]
```diff
@@ -100,6 +100,12 @@
             "default": true,
             "tip": "Associate public IP (VPC only)"
           },
+          "provider_hostname": {
+            "label": "Use provider-specific hostname (overrides DNS suffix)",
```
Comment (dereklwood) [inline lib/provisioner/worker/plugins/providers/fog_provider/fog_provider.json]: I suggest "provider-generated" would be more clear

**Comment by dereklwood** [comment]
i believe this will not prevent the hostname from getting changed here: https://github.com/caskdata/coopr-provisioner/blob/hotfix/coopr-721-provider-hostnames/lib/provisioner/worker/plugins/providers/fog_provider/lib/fog_provider/google.rb#L186

**Comment by dereklwood** [comment]
Also, no joyent, etc?

**Comment by dereklwood** [comment]
oops, didn't mean to close...reopening

**Comment by duddun** [comment]
LGTM -- only question is (same as above) why not rackspace, etc.

**Comment by dereklwood** [comment]
:+1:

**Pull request merged**